.gitignore

```
@@ -1,21 +1,54 @@
-node_modules
-coverage
-.nyc_output
-.DS_Store
-*.log
-.vscode
-.idea
-compiled
-.awcache
-.rts2_cache*
-*.proto
-example/js/vendor
-dist
-.rpt2_*
-.cache
-.npmrc
-junit.xml
-public
-examples/sandbox/public
-tsconfig.tsbuildinfo
-.history/
\ No newline at end of file
+*
+
+# directories
+!/.circleci
+!/.circleci/**
+!/.github/
+!/.github/**
+!/docs/
+!/docs/**
+
+!/examples/
+!/examples/*
+!/examples/sandbox/*
+!/examples/sandbox/json/
+!/examples/sandbox/json/**
+
+!/integration/
+!/integration/*
+!/integration/src/
+!/integration/src/**
+
+!/packages/
+!/packages/*
+!/packages/**/src/
+!/packages/**/src/**
+!/packages/**/.npmignore
+!/packages/**/jest.config.js
+!/packages/**/package.json
+!/packages/**/tsconfig.json
```

```suggestion
!/packages/*/src/
!/packages/*/src/**
!/packages/*/.npmignore
!/packages/*/jest.config.js
!/packages/*/package.json
!/packages/*/tsconfig.json
```
(haven't tested this yet, but we probably don't want to apply these recursively)